dags/analyzer_helper/discourse/extract_raw_info.py

```diff
@@ -0,0 +1,275 @@
+        self.platform_db = self.client[platform_id]
+        self.rawmemberactivities_collection = self.platform_db["rawmemberactivities"]
+
+    def close(self):
+        """
+        Close the Neo4j connection.
+        """
+        self.driver.close()
+
+    def fetch_post_details(self, created_at: Optional[str] = None, comparison: Optional[str] = None) -> list:
+        """
+        Fetch details of posts from the Discourse forum.
+
+        :param created_at: Optional datetime string to filter posts created after this date.
+        :param comparison: Optional comparison operator, either 'gt' for greater than or 'gte' for greater than or equal to.
+        :return: List of dictionaries containing post details.
+        """
+        if comparison:
+            assert comparison in {'gt', 'gte'}, "comparison must be either 'gt' or 'gte'"
+
+        where_clause = ""
+        if created_at and comparison:
+            operator = '>' if comparison == 'gt' else '>='
+            where_clause = f"WHERE post.createdAt {operator} $createdAt"
+        # With neo4j automatically generated ids
+        # query = f"""
+        # MATCH (forum:DiscourseForum {{endpoint: $forum_endpoint}})
+        # WITH forum
+        # MATCH (topic:DiscourseTopic {{forumUuid: forum.uuid}})
+        # MATCH (topic)-[:HAS_POST]->(post:DiscoursePost)
+        # {where_clause}
+        # OPTIONAL MATCH (post)<-[:POSTED]-(author:DiscourseUser)
+        # OPTIONAL MATCH (post)<-[:LIKED]-(likedUser:DiscourseUser)
+        # OPTIONAL MATCH (post)-[:REPLY_TO]->(repliedPost:DiscoursePost)
+        # OPTIONAL MATCH (repliedPost)<-[:POSTED]-(repliedAuthor:DiscourseUser)
+        # RETURN
+        # id(post) AS post_id,
+        # id(author) AS author_id,
+        # post.createdAt AS created_at,
+        # author.name AS author_name,
+        # collect(DISTINCT likedUser.id) AS reactions,
+        # id(repliedPost) AS replied_post_id,
+        # id(repliedAuthor) AS replied_post_user_id,
+        # id(topic) AS topic_id
+        # LIMIT 10
+        # """
+        # Without neo4j automatically generated ids
+        query = f"""
+        MATCH (forum:DiscourseForum {{endpoint: $forum_endpoint}})
+        WITH forum
+        MATCH (topic:DiscourseTopic {{forumUuid: forum.uuid}})
+        MATCH (topic)-[:HAS_POST]->(post:DiscoursePost)
+        {where_clause}
+        OPTIONAL MATCH (post)<-[:POSTED]-(author:DiscourseUser)
+        OPTIONAL MATCH (post)<-[:LIKED]-(likedUser:DiscourseUser)
+        OPTIONAL MATCH (post)-[:REPLY_TO]->(repliedPost:DiscoursePost)
+        OPTIONAL MATCH (repliedPost)<-[:POSTED]-(repliedAuthor:DiscourseUser)
+        RETURN
+        post.id AS post_id,
+        author.id AS author_id,
+        post.createdAt AS created_at,
+        author.name AS author_name,
+        collect(DISTINCT likedUser.id) AS reactions,
+        repliedPost.id AS replied_post_id,
+        repliedAuthor.id AS replied_post_user_id,
+        topic.id AS topic_id
+        LIMIT 10
```

Comment:
Why we're limiting the return data? It could prevent many of the data from being extracted.